libgeopmd/src/DerivativeSignal.cpp

```diff
@@ -26,6 +26,7 @@ namespace geopm
         , m_is_batch_ready(false)
         , m_sleep_time(sleep_time)
         , m_last_result(NAN)
+        , m_nan_replace(0)
```

Comment:
Since this is intended for only 2 signals, should the default behavior match the current behavior? Then we could just apply the zero-fill for the two signals where we know that's desirable.

Reply:
Other signals are currently using MSR::TIME which is guaranteed to monotonically increase with each sample. When new signals are introduced that use timers that might not increase with each read, we will add a new constructor that allows m_nan_replace to be declared.